File: docs-examples/openai-sdk-python-chat/basic.py
import os
from dotenv import load_dotenv
from openai import OpenAI

load_dotenv(override=True)

if not os.getenv('VERLON_API_KEY') or not os.getenv('VERLON_GATE_ID'):
    print('Error: VERLON_API_KEY and VERLON_GATE_ID environment variables are required')
    print('Copy .env.example to .env and fill in your credentials')
    exit(1)

client = OpenAI(
    base_url="https://api.verlon.ai/v1",
    api_key=os.getenv('VERLON_API_KEY'),
)

response = client.chat.completions.create(
    model=os.getenv('VERLON_GATE_ID'),
    messages=[
        {"role": "user", "content": "Hello! Tell me a fun fact about space."}
    ]
)

print('Chat completion successful!')
print('Response:', response.choices[0].message.content)
print('Model:', response.model)
print('Usage:', response.usage)

File: docs-examples/sdk-chat/basic.ts
import { config } from 'dotenv';
import OpenAI from 'openai';

config({ path: '.env', override: true });

if (!process.env.VERLON_API_KEY || !process.env.VERLON_GATE_ID) {
  console.error('Error: VERLON_API_KEY and VERLON_GATE_ID environment variables are required');
  console.error('Copy .env.example to .env and fill in your credentials');
  process.exit(1);
}

const openai = new OpenAI({
  apiKey: process.env.VERLON_API_KEY,
  baseURL: 'https://api.verlon.ai/v1',
});

// Basic chat example from documentation
const response = await openai.chat.completions.create({
  model: process.env.VERLON_GATE_ID,  // Gate UUID as model
  messages: [
    { role: 'system', content: 'You are a helpful assistant.' },
    { role: 'user', content: 'Explain quantum computing in simple terms' }
  ],
});

console.log(response.choices[0].message.content);
console.log('Model:', response.model);
console.log('Usage:', response.usage);
